crowbar_framework/app/models/node_object.rb

```diff
@@ -1585,16 +1585,25 @@ def unique_device_for(device)
       end
       result = disk_lookups.map do |type|
         if meta["disks"][type] and not meta["disks"][type].empty?
-          "#{type}/#{meta["disks"][type].first}"
+          meta["disks"][type]
         end
       end
-
+      unless result.empty?
+        result = result[0]
+        match = result.find { |b| b =~ /^wwn-/ } ||
+          result.find { |b| b =~ /^scsi-[a-zA-Z]/ } ||
+          result.find { |b| b =~ /^scsi-[^1]/ } ||
+          result.find { |b| b =~ /^scsi-/ } ||
+          result.find { |b| b =~ /^ata-/ } ||
+          result.find { |b| b =~ /^cciss-/ } ||
+          result.first
+      end
       # virtio disk might have neither by-path nor by-id links, use the /dev/vdX
       # name in that case
-      if result.empty?
+      if match.empty?
```

Comment:
Isn't it possible that `match` is not defined (in case we don't enter in the `unless result.empty?` branch)?